### PR TITLE
autopep and manual changes from pylint/flake

### LINKED ISSUE #14: Make code conform to PEP8 style
```
flake8 . --count  --statistics
...
6     E101 indentation contains mixed spaces and tabs
1     E111 indentation is not a multiple of four
1     E116 unexpected indentation (comment)
5     E117 over-indented
2     E122 continuation line missing indentation or outdented
2     E124 closing bracket does not match visual indentation
3     E125 continuation line with same indent as next logical line
20    E127 continuation line over-indented for visual indent
48    E128 continuation line under-indented for visual indent
25    E201 whitespace after '('
25    E202 whitespace before ')'
1     E222 multiple spaces after operator
2     E225 missing whitespace around operator
8     E231 missing whitespace after ','
4     E265 block comment should start with '# '
1     E271 multiple spaces after keyword
1     E272 multiple spaces before keyword
15    E302 expected 2 blank lines, found 1
7     E303 too many blank lines (2)
1     E305 expected 2 blank lines after class or function definition, found 1
1     E401 multiple imports on one line
1     E402 module level import not at top of file
105   E501 line too long (83 > 79 characters)
11    E502 the backslash is redundant between brackets
3     E711 comparison to None should be 'if cond is not None:'
1     E712 comparison to True should be 'if cond is True:' or 'if cond:'
1     E713 test for membership should be 'not in'
23    F401 '.base.BIGINT' imported but unused
2     F811 redefinition of unused 'DECIMAL' from line 28
1     F841 local variable 'query' is assigned to but never used
3     W191 indentation contains tabs
1     W291 trailing whitespace
1     W292 no newline at end of file
20    W293 blank line contains whitespace
1     W601 .has_key() is deprecated, use 'in'
3     W605 invalid escape sequence '\s'
356
```
See [flake8-full.txt](https://github.com/IBM/sqlalchemy-ibmi/files/4067464/flake8-full.txt) for full details.

We can use [autopep8](https://github.com/hhatto/autopep8) to fix a lot of these issues.

### PR DESCRIPTION
**Pull request by naveen-ram**
Fixes #14

### CODE REVIEW

File: sqlalchemy_ibmi/backend.py [hunk 1]
```diff
@@ -88,14 +106,46 @@ def create_connect_args(self, url):
...
+    def do_recover_twophase(self, connection):
+        pass
+
+    def get_isolation_level(self, dbapi_conn):
+        pass
+
     supports_unicode_statements = True
     supports_sane_rowcount = False
     supports_sane_multi_rowcount = False
-    supports_native_decimal = True
```
Comment (kadler) [inline sqlalchemy_ibmi/backend.py]: Looks like this was duplicated previously. The question is: which value _should_ it have? I suggest we add a TODO to investigate further. I can't find much on what this does, other than it is used by the Oracle and possibly MySQL interfaces.

Certainly, Db2 has a native DECIMAL type _and_ the driver handles it, but this setting seems to be about something a little different so I'm not sure.

File: sqlalchemy_ibmi/backend.py [hunk 2]
```diff
@@ -105,25 +155,31 @@ def create_connect_args(self, url):
...
+                connectors = [
+                    "DRIVER={%s}" %
+                    keys.pop(
+                        'driver',
+                        self.pyodbc_driver_name),
+                    'System=%s' %
+                    keys.pop(
+                        'host',
+                        ''),
+                    'DBQ=QGPL']
```
Comment (kadler) [inline sqlalchemy_ibmi/backend.py]: The auto formatting on this did quite the horrific job, but we need to refactor this code anyway, so I'm ok with it.

File: setup.py [hunk 3]
```diff
@@ -4,7 +4,11 @@
...
 import re
 
-v = open(os.path.join(os.path.dirname(__file__), 'sqlalchemy_ibmi', '__init__.py'))
+v = open(
+    os.path.join(
+        os.path.dirname(__file__),
+        'sqlalchemy_ibmi',
+        '__init__.py'))
 VERSION = re.compile(r".*__version__ = '(.*?)'", re.S).match(v.read()).group(1)
 v.close()
```
[the remark this answers is not in the document]
Comment (kadler) [reply setup.py]: Maybe put a TODO?

File: sqlalchemy_ibmi/base.py [hunk 4]
```diff
@@ -19,165 +19,206 @@
...
-    'unnest', 'element', 'percentile_disc', 'upper', 'exec', 'power', 'var_pop', 'exp',
-    'real', 'var_samp', 'false', 'recursive', 'varchar', 'filter', 'ref', 'varying',
-    'float', 'regr_avgx', 'width_bucket', 'floor', 'regr_avgy', 'window', 'fusion',
-    'regr_count', 'within', 'asc'])
-
-
-class _IBM_Boolean(sa_types.Boolean):
-
-    def result_processor(self, dialect, coltype):
+RESERVED_WORDS = {'activate', 'disallow', 'locale', 'result', 'add',
```
Comment (kadler) [inline sqlalchemy_ibmi/base.py]: Does it really format things this way instead of putting things on a new line and leaving them less indented?
Personally, I'd prefer these all go on separate lines to make diffs easier when things are changed/added. This will significantly increase the number of lines, though. We could mitigate that somewhat by putting it in its own module and just importing it here.

File: sqlalchemy_ibmi/base.py [hunk 5]
```diff
@@ -19,165 +19,206 @@
...
                 return None
             if isinstance(value, datetime.datetime):
                 value = datetime.date(value.year, value.month, value.day)
             return str(value)
+
         return process
 
+
 class DOUBLE(sa_types.Numeric):
+    """IBM i DB2 Double class"""
```
Comment (kadler) [inline sqlalchemy_ibmi/base.py]:
```suggestion
    """Represents a Db2 DOUBLE column"""
```
These could all follow this same pattern

File: sqlalchemy_ibmi/base.py [hunk 6]
```diff
@@ -205,51 +246,51 @@ class XML(sa_types.Text):
 
 
 class DB2TypeCompiler(compiler.GenericTypeCompiler):
+    """IBM i DB2 Type Compiler"""
```
Comment (kadler) [inline sqlalchemy_ibmi/base.py]:
```suggestion
    """IBM i Db2 Type Compiler"""
```

The marketing people changed the brand from DB2 -> Db2 a couple years ago. It's pretty dumb, but might as well use the correct branding for any new code...

File: sqlalchemy_ibmi/base.py [hunk 7]
```diff
@@ -261,102 +302,103 @@ def visit_VARGRAPHIC(self, type_):
...
     def visit_now_func(self, fn, **kw):
         return "CURRENT_TIMESTAMP"
 
-    def for_update_clause(self, select):
-        if select.for_update == True:
+    def for_update_clause(self, select, **kw):
+        if select.for_update:
             return ' WITH RS USE AND KEEP UPDATE LOCKS'
-        elif select.for_update == 'read':
+        if select.for_update == 'read':
```
Comment (kadler) [inline sqlalchemy_ibmi/base.py]: As written, the second if will never happen since 'read' is _truthy_ will be selected by the first if condition. These conditions should be switched around.

File: sqlalchemy_ibmi/backend.py [hunk 8]
```diff
@@ -88,14 +106,46 @@ def create_connect_args(self, url):
         return [[";".join(connectors)], connect_args]
 
 
-class AS400Dialect_pyodbc(PyODBCConnector, DB2Dialect):
+class AS400DialectPyodbc(PyODBCConnector, DB2Dialect):
+    """ IBMi pyodbc sqlalchemy dialect """
+
+    def get_temp_table_names(self, connection, schema=None, **kw):
+        pass
```
[the remark this answers is not in the document]
Comment (naveen-ram) [reply sqlalchemy_ibmi/backend.py]: these methods are from the dialect interface but didn't have an implementation in the ibmdb_sa backend. Pylint complained that they weren't overridden, but there is no parent class, other than the NotImplementedError

File: sqlalchemy_ibmi/backend.py [hunk 9]
```diff
@@ -88,15 +106,47 @@ def create_connect_args(self, url):
...
     supports_unicode_statements = True
     supports_sane_rowcount = False
     supports_sane_multi_rowcount = False
-    supports_native_decimal = True
-    supports_char_length = True
     supports_native_decimal = False
-
-    pyodbc_driver_name = "iSeries Access ODBC Driver"
+    supports_char_length = True
+    pyodbc_driver_name = "IBM i Access ODBC Driver"
```
Comment (kadler) [inline sqlalchemy_ibmi/backend.py]: Please do this change in a separate PR. Changing the driver name is not a linter change.

File: sqlalchemy_ibmi/base.py [hunk 10]
```diff
@@ -19,165 +19,206 @@
...
             if value is None:
                 return None
-            elif bool(value):
+            if bool(value):
                 return '1'
-            else:
-                return '0'
+            return '0'
+
         return process
```
[the remark this answers is not in the document]
Comment (kadler) [reply sqlalchemy_ibmi/base.py]: Nevermind, `bind_processoris returning the `process` function it just defined.

File: sqlalchemy_ibmi/backend.py [hunk 11]
```diff
@@ -138,13 +190,15 @@ def create_connect_args(self, url):
...
+                connectors.append(
+                    "AutoTranslate=%s" %
+                    keys.pop("odbc_autotranslate"))
 
             connectors.extend(['%s=%s' % (k, v) for k, v in keys.items()])
         return [[";".join(connectors)], connect_args]
 
 
-dialect = DB2Dialect_pyodbc
+dialect = AS400DialectPyodbc
```
Comment (kadler) [inline sqlalchemy_ibmi/backend.py]: Let's do this in a separate PR as well.
